pkg/performance/collector.go

```diff
@@ -51,11 +52,20 @@ type NewContinuousCollector func(logr.Logger, CollectionConfig) (ContinuousColle
 type NewPointCollector func(logr.Logger, CollectionConfig) (PointCollector, error)
 
 type CollectorCapabilities struct {
-	SupportsOneShot    bool
-	SupportsContinuous bool
-	RequiresRoot       bool
-	RequiresEBPF       bool
-	MinKernelVersion   string
+	SupportsOneShot      bool
+	SupportsContinuous   bool
+	RequiredCapabilities []capabilities.Capability
+	MinKernelVersion     string
+}
+
+// CanRun checks if the collector can run with current process capabilities
+func (c CollectorCapabilities) CanRun() (bool, []capabilities.Capability, error) {
```

Comment:
it'll just be easier if you made it `CanRun() error`. and the error is something like
```golang
errors.Join(
   fmt.Errorf("missing capability: %v", capability),
   fmt.Errorf("missing capability: %v", capability),
)
```

if you print the error outside it will include all of them